src/Model/Behavior/MessagesBehavior.php

```diff
@@ -0,0 +1,45 @@
+<?php
+namespace CakeMojo\Messages\Model\Behavior;
+
+use Cake\ORM\Behavior;
+use Cake\ORM\Table;
+use Cake\ORM\TableRegistry;
+use Cake\Utility\Hash;
+
+/**
+ * MessagesBehavior behavior
+ */
+class MessagesBehavior extends Behavior
+{
+
+    /**
+     * Default configuration.
+     * @var id //of the table for pass like foreign key
+     * @array data
+     * -subject
+     * -body
+     * -model //you can to assign table name or get automatically from model
+     * @var array
+     */
+    protected $_defaultConfig = [
+        'subject' => 'Message subject',
+        'body' => 'This is the body'
+    ];
+
+    public function addMessage($id, array $data){
+        $config = $this->_defaultConfig;
+        Hash::merge($config, $data);
+
+        $messagesTable = TableRegistry::get('CakeMojo/Messages.Messages');
+
+        $message = $messagesTable->newEntity();
+
+        $message->foreign_key = $id;
+        $message->subject = $data['subject'];
```

Comment:
what if there is no 'subject' key, we need to check it

Reply:
Ok, in check function for this parameter if not exist, I show Exception error or subject string by default, is not necessary check if I pass a default configuration.

Reply:
@chav170 resolved the problem (Magic Hash::get)